concrete/blocks/core_area_layout/controller.php

```diff
@@ -276,48 +279,52 @@ public function save($post)
     }
 
     /**
-     * @param \SimpleXMLElement $blockNode
-     * @param mixed $page
+     * {@inheritdoc}
      *
-     * @return array<string,mixed>
+     * @see \Concrete\Core\Block\BlockController::getImportData()
      */
     public function getImportData($blockNode, $page)
     {
+        if (!isset($blockNode->arealayout)) {
+            return [];
+        }
+        $node = $blockNode->arealayout;
+        $type = (string) $node['type'];
         $args = [];
-        if (isset($blockNode->arealayout)) {
-            $type = (string) $blockNode->arealayout['type'];
-            $node = $blockNode->arealayout;
-            switch ($type) {
-                case 'theme-grid':
-                    $args['gridType'] = 'TG';
-                    $args['arLayoutMaxColumns'] = (string) $node['columns'];
-                    $args['themeGridColumns'] = (int) (count($node->columns->column));
-                    $args['offset'] = [];
-                    $args['span'] = [];
-                    $i = 0;
-                    foreach ($node->columns->column as $column) {
-                        $args['span'][$i] = (int) ($column['span']);
-                        $args['offset'][$i] = (int) ($column['offset']);
-                        $i++;
-                    }
-                    break;
-                case 'custom':
-                    $args['gridType'] = 'FF';
-                    $args['isautomated'] = true;
-                    $args['spacing'] = (int) ($node['spacing']);
-                    $args['columns'] = (int) (count($node->columns->column));
-                    $customWidths = (int) ($node['custom-widths']);
-                    if ($customWidths == 1) {
-                        $args['isautomated'] = false;
-                    }
-                    $args['width'] = [];
-                    $i = 0;
+        switch ($type) {
+            case 'theme-grid':
+                $args += [
+                    'gridType' => 'TG',
+                    'arLayoutMaxColumns' => (string) $node['columns'],
+                    'themeGridColumns' => count($node->columns->column),
+                    'offset' => [],
+                    'span' => [],
+                ];
+                foreach ($node->columns->column as $column) {
+                    $args['span'][] = (int) $column['span'];
+                    $args['offset'][] = (int) $column['offset'];
+                }
+                break;
+            case 'custom':
+                $preset = $this->resolveLayoutPreset(isset($node['preset']) ? (string) $node['preset'] : '', $page);
+                if ($preset !== null) {
+                    $args += [
+                        'gridType' => $preset->getIdentifier(),
+                        'arLayoutPresetID' => $preset->getIdentifier(),
+                    ];
+                } else {
+                    $args += [
```

Comment:
I'm pretty sure we don't need these values if we are importing a preset, but I didn't test the code.
On Monday I'll do a deep test on what's actually needed.

Reply:
👍

Reply:
I just checked the values used when importing a preset, and we need only `gridType` and `arLayoutPresetID`.

See:

https://github.com/concrete5/concrete5/blob/05834082152c4fc6b900bd0b4b2d0b780c976865/concrete/blocks/core_area_layout/form.php#L23-L47

and

https://github.com/concrete5/concrete5/blob/05834082152c4fc6b900bd0b4b2d0b780c976865/concrete/blocks/core_area_layout/controller.php#L337-L379

(the `getImportData` method basically converts the XML data to an array that's compatible with the data received via POST, and this data is used in the `addFromPost` method both for XML and web POST)